src/storage/sql_table.cpp

```diff
@@ -145,7 +156,20 @@ bool SqlTable::Select(transaction::TransactionContext *const txn, const TupleSlo
   bool result = old_dt_version.data_table->Select(txn, slot, out_buffer);
   std::memcpy(out_buffer->ColumnIds(), original_column_ids, sizeof(col_id_t) * out_buffer->NumColumns());
 
-  // TODO(Yashwanth): handle default values
+  // Get the DefaultValueMap visible to this transaction
+  DefaultValueMap default_val_map = curr_dt_version.default_value_map;
+
+  // Populate default values into the empty columns of the ProjectedRow (if any)
```

Comment:
See the comment in `UpdateSchema`.  This is more complicated then just checking for `NULL` since default values are only retroactive if they are applied when a column is first added.  Additionally, you can override a default value by explicitly setting the value to `NULL` (if constraints allow) in the insert statement.

Reply:
The defaultValueMap that I'm constructing is assumed to contain only the values that should be applied retroactively. I've assumed that the catalog maintains the default values that are applied after the column creation at another place.

If the default value in the `Column` class is the only copy, then I need to maintain a flag indicating whether this default value was assigned during creation or later.